Cargill/splinter-ui
# Add circuit stats and input filter

This PR adds: 
 - Logic to fetch the local node ID from the splinter API
 - Logic to process circuit and proposal data coming from the API
 - Mock data for circuits and proposals
 - Adds an input filter for filtering circuits shown
 - Adds summary stats to the top of the page

#### To test

1. Run 
```
docker-compose -f docker/docker-compose.yaml up --build
```
2. To see the alpha node UI go to: `localhost:3030`

3. Click on the `circuits` option in the left side menu, it will forward you to  `localhost:3030/circuits`: 
<img width="1196" alt="Screen Shot 2020-04-14 at 1 53 40 PM" src="https://user-images.githubusercontent.com/14094978/79262730-603bae80-7e57-11ea-9b73-e046df5dfeba.png">

4. Test the filter input. You can filter by ID, management type, service type, members and comments. The full data can be found at `saplings/circuits/src/mockData/mockCircuits.js` and `saplings/circuits/src/mockData/mockProposals.js`.

5. To see the beta node UI go to: `localhost:3031`

6. Click on the `circuits` option in the left side menu, it will forward you to  `localhost:3031/circuits`: 
 
<img width="1196" alt="Screen Shot 2020-04-14 at 2 13 54 PM" src="https://user-images.githubusercontent.com/14094978/79264471-4059ba00-7e5a-11ea-9d55-f7c95dee371f.png">


Notice that  in the beta node UI the count of circuits with action required shows `3` while in the alpha node it shows `2`. That's because that are 3 circuits proposals the beta node needs to vote on, and 2 circuits proposals that the alpha node needs to vote on. 

P.S This PR seems big, but 600+ lines are just mock data added for tests purposes and will be removed soon. They don't require careful review.
